### PR TITLE
[Edric]-[Search with optional]

### PR DESCRIPTION
**Pull request by Edrlc**
- Added functionality for searching with optional words

### CODE REVIEW

File: server/utils/search-optional.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+            })
+        }
+        // remove duplicates
+        return [...new Set(newResults)];
+    } else {
+        return results;
+    }
+}
+
+function removeWhiteSpaceFrom(string) {
```
Comment (marcusmariette) [inline server/utils/search-optional.js:47]: we can probably create a ```searchUtils.js``` file for stuff like this, I'll handle it when I work on the app next, just for future reference.

File: server/server.js [hunk 2]
```diff
@@ -92,12 +93,9 @@ app.get('/keywords', async (req, res) => {
...
     if (query !== undefined) {
         let results = [];
-        let matchOR = query.match(/\w+\/\w+/g);
-        if (matchOR != null) {
-            // search here
-            results = searchWithOr(query, matchOR, searchWithOr, results);
-        }
 
+        results = searchWithOr(query, searchWithOr, results);
+        results = searchWithOptional(query, results);
```
Comment (marcusmariette) [inline server/server.js:98]: I assume that this does not override the other results, but adds to them if we have any??
Does that mean that we are running every single possible search on each query even if there is not the identifier in there?

**Review by marcusmariette — COMMENTED** [review]
Few Notes

File: server/server.js (hunk 2 above)
Comment (Edrlc) [reply server/server.js:98]: yeah it adds to the results, but there's a check inside to just return back the result if it doesn't match the regex

Comment (marcusmariette) [reply server/server.js:98]: awesome!

**Review by marcusmariette — APPROVED** [review]

**Pull request merged**